ee/tabby-webserver/src/oauth/github.rs

```diff
@@ -0,0 +1,48 @@
+use std::sync::Arc;
+use anyhow::Result;
+use hyper::{Method, Request};
+use serde::{Deserialize, Serialize};
+use crate::schema::auth::OAuthProvider;
+use crate::schema::ServiceLocator;
+
+#[derive(Serialize, Deserialize)]
+pub struct GithubOAuthResponse {
+    #[serde(default)]
+    access_token: String,
+    #[serde(default)]
+    scope: String,
+    #[serde(default)]
+    token_type: String,
+
+    #[serde(default)]
+    error: String,
+    #[serde(default)]
+    error_description: String,
+    #[serde(default)]
+    error_uri: String,
+}
+
+pub async fn access_token(
```

Comment:
I feel we should just read the email directly from github to handle the "authentication", what do you think? (it can be done in a followup PR) after acquiring the access token.